packages/tscc/src/shared/sourcemap_splice.ts

```diff
@@ -9,68 +9,62 @@ import {SourceMapConsumer, SourceMapGenerator, Mapping, RawSourceMap} from 'sour
  */
 export default async function spliceSourceMap(content: string, map: RawSourceMap, spliceIntervals: [number, number][]): Promise<RawSourceMap> {
 	const consumer = await new SourceMapConsumer(map);
-	const generator = new SourceMapGenerator({file: map.file});
+	const generator = new SourceMapGenerator({ file: map.file });
 
 	const seeker = new Seeker(content, spliceIntervals);
 
-	consumer.eachMapping(({source, generatedLine, generatedColumn, originalLine, originalColumn, name}) => {
+	consumer.eachMapping(({ source, generatedLine, generatedColumn, originalLine, originalColumn, name }) => {
 		// line numbers in mozilla/source-map are 1-based. column numbers are 0-based.
 		seeker.seek(generatedLine - 1, generatedColumn);
 		if (seeker.isInInterval()) return;
 		let transformedLine = seeker.getTransformedLine();
 		let transformedColumn = seeker.getTransformedColumn();
 		const mapping = getMapping(
-			source, transformedLine + 1, transformedColumn,
-			originalLine, originalColumn, name
+			source, 
+			transformedLine + 1, 
+			transformedColumn,
+			originalLine, 
+			originalColumn,
+			name
 		);
 		generator.addMapping(mapping);
 	});
 
 	return generator.toJSON();
 }
 
-const mapping: Mapping = {
-	source: undefined,
-	generated: {
-		line: undefined,
-		column: undefined
-	},
-	original: {
-		line: undefined,
-		column: undefined
-	},
-	name: undefined
-};
-
-const mappingWithoutOriginal = <Mapping>{
-	source: undefined,
-	generated: {
-		line: undefined,
-		column: undefined
-	}
-};
-
-function getMapping(source, generatedLine, generatedColumn, originalLine, originalColumn, name): Mapping {
-	if (typeof originalLine !== 'number' && typeof originalLine !== 'number') {
-		mappingWithoutOriginal.source = source;
-		mappingWithoutOriginal.generated.line = generatedLine;
-		mappingWithoutOriginal.generated.column = generatedColumn;
-		return mappingWithoutOriginal;
-	}
-	mapping.source = source;
-	mapping.generated.line = generatedLine;
-	mapping.generated.column = generatedColumn;
-	mapping.original.line = originalLine;
-	mapping.original.column = originalColumn;
-	mapping.name = name;
-	return mapping;
+/**

```

Comment:
The above function implements object pooling, in order to reduce garbage collection of V8. Unless it can be proven that this does not significantly affect performance, I suggest not to remove it.

Reply:
tsc hated it so I had to modify it, and it seemed sinful as hell anyway. I would need your help to re-insert it in a way that compiles.